css/amp.css

```diff
@@ -876,6 +876,8 @@ amp-autocomplete > input,
 .i-amphtml-autocomplete-results {
   font-size: 1rem; /* default-ui-font-size */
   line-height: 1.5rem; /* default-ui-space-large */
+  max-height: 40vh;
```

Comment:
Can you please explain why the value for `max-height` is `40vh` specifically? I think we discussed the sizes of various mobile keyboards a little while ago, but it'd be good to have a link cited here for future authors' benefit

Reply:
One primary goal of this change is to make it so that (in tandem with switching the autocomplete to point up or down depending on its position in the viewport, which is not part of this PR) the results list never adds height to either the top of bottom of the page.

Consider results whose natural size is 80vh: if the autocomplete is in the top half of the last viewport, those results will now extend 40vh downwards, which will be within the existing page boundaries (since by definition there are at least 50vh below an element in the top half of the viewport).

We choose 40vh (instead of 50vh) to fudge it a little bit - for visual design reasons we want some margin between the bottom of the results and the bottom of the viewport, and it also helps to ensure we still remain within the viewport in cases where we measure slightly wrong (if it were 50vh, the code would have to be really anal about getting the position of the bottom of the input in the down case and the top of the input in the up case).

In the software keyboard case, the presence/absence of the keyboard doesn't change the viewport (likely for performance reasons?) so that won't change the above goal.

It WILL however affect our secondary goal of keeping all results on screen at once. Part of that goal is aesthetic (see margin comment above), but it also means that scrolling the autocomplete results keeps the input visible on screen at the same time (since scrolling will happen within the results div rather than within the page). With an arbitrary-height software keyboard we lose the aesthetic benefit, but keep (most of) the field-visibility benefit, since tap-drags within the visible part of results div will still scroll that in place.

I think a bigger issue is that unless the field is moved to the top of the viewport when the mobile keyboard pops up, it's possible that the user won't notice that there's a results div underneath it, but this PR doesn't change that behavior. That's something we should think about/fix when implementing the logic to make the results point up or down.

Reply:
Ok I think I understand, thanks. It could help to have a comment in the CSS next to the `max-height: 40vh;` rule that summarizes and documents your reasoning